### PR TITLE
manifest: adapt samples to new ot cb register api

### PR DESCRIPTION
**Pull request by pjasinski990**
Use new OpenThread callback register api
(https://github.com/nrfconnect/sdk-zephyr/pull/1030).

Signed-off-by: Piotr Jasiński <piotr.jasinski@nordicsemi.no>

### CODE REVIEW

**Comment by canisLupus1313** [comment]
You should add `manifest:` in the commit message.

**Review by canisLupus1313 — APPROVED** [review]

**Review by edmont — APPROVED** [review]

**Review by mariuszpos — APPROVED** [review]

**Comment by rlubos** [comment]
Please sort out the twister failures, the PR is blocking other sdk-zephyr PRs.

File: samples/openthread/cli/src/low_power.c [hunk 1]
```diff
@@ -29,8 +28,11 @@ static void on_thread_state_changed(uint32_t flags, void *context)
 		}
 	}
 }
+static struct openthread_state_changed_cb ot_state_chaged_cb = {
```
Comment (edmont) [inline samples/openthread/cli/src/low_power.c:31]: Missing blank line here.

**Comment by pjasinski990** [comment]
> Please sort out the twister failures, the PR is blocking other sdk-zephyr PRs.

Should be ok now, merging once CI is green

**Pull request merged**